### PR TITLE
CPDP-2694: handle amended returns

### PR DESCRIPTION
**Pull request by mikail-khan**

### CODE REVIEW

File: app/uk/gov/hmrc/cgtpropertydisposals/repositories/returns/AmendReturnsRepository.scala [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+      collectionName = "amend-returns",
+      mongo = component.mongoConnector.db,
+      domainFormat = SubmitReturnRequest.format
+    )
+    with AmendReturnsRepository
+    with CacheRepository[SubmitReturnRequest] {
+
+  //TODO: review parameters
+  val cacheTtl: FiniteDuration  = config.underlying.get[FiniteDuration]("mongodb.draft-returns.expiry-time").value
+  val maxDraftReturns: Int      = config.underlying.get[Int]("mongodb.draft-returns.max-draft-returns").value
```
Comment (mikail-khan) [inline app/uk/gov/hmrc/cgtpropertydisposals/repositories/returns/AmendReturnsRepository.scala:63]: @AndyHWChung are these parameters applicable? How should the house-keeping be done for the entries in this collection?

Comment (AndyHWChung) [reply app/uk/gov/hmrc/cgtpropertydisposals/repositories/returns/AmendReturnsRepository.scala:63]: with the way this is being used there shouldn't be a limit to the number of  returns returned for a cgt reference - to correctly display the "recently amended" content we have to know for certain whether the return was recently amended which means looking at everything in this collection for the cgt reference.

The `cacheTtle` needs to be separate from the one for draft returns. The draft return ttl is 30 days, but this ttl needs to be 24 hours. This ttl controls how long a user will see the "recently amended" version of the content on the homepage for after the return has been amended.

File: test/uk/gov/hmrc/cgtpropertydisposals/repositories/returns/AmendReturnsRepositorySpec.scala [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+  val cgtReference = sample[CgtReference]
+
+  "DraftAmendReturnsRepository" when {
+    "inserting" should {
+      "create a new draft return successfully" in {
+        val submitReturnRequest = sample[SubmitReturnRequest]
+        await(repository.save(submitReturnRequest, cgtReference).value) shouldBe Right(())
+      }
+    }
+    "getting"   should {
```
Comment (AndyHWChung) [inline test/uk/gov/hmrc/cgtpropertydisposals/repositories/returns/AmendReturnsRepositorySpec.scala:50]: could we store two sample `SubmitReturnRequest` with the same cgt referebce in this test to make sure we get both back?

File: app/uk/gov/hmrc/cgtpropertydisposals/repositories/returns/AmendReturnsRepository.scala [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+  override def fetch(cgtReference: CgtReference): EitherT[Future, Error, List[SubmitReturnRequest]] =
+    EitherT(
+      preservingMdc {
+        get(cgtReference)
+      }
+    )
+
+  override def save(
+    submitReturnRequest: SubmitReturnRequest,
+    cgtReference: CgtReference
```
Comment (AndyHWChung) [inline app/uk/gov/hmrc/cgtpropertydisposals/repositories/returns/AmendReturnsRepository.scala:76]: `cgtReference` isn't being used here

Comment (mikail-khan) [reply app/uk/gov/hmrc/cgtpropertydisposals/repositories/returns/AmendReturnsRepository.scala]: nasty - the compiler should have flagged that (and Intellij!) and failed compilation

**Comment by AndyHWChung** [comment]
LGTM

**Pull request merged**